File: src/main/java/com/ptaf/ai/model/AiGenerationMode.java
package com.ptaf.ai.model;

import java.util.Locale;

public enum AiGenerationMode {
    PREVIEW,
    WRITE,
    STRICT;

    public static AiGenerationMode fromString(String value) {
        if (value == null || value.isBlank()) {
            return PREVIEW;
        }
        return switch (value.trim().toUpperCase(Locale.ROOT)) {
            case "PREVIEW" -> PREVIEW;
            case "WRITE" -> WRITE;
            case "STRICT" -> STRICT;
            default -> throw new IllegalArgumentException("Unsupported mode: " + value + ". Use preview|write|strict");
        };
    }
}

File: ptaf/ai/model/ai_generation_mode.py
from __future__ import annotations

from enum import Enum


class AiGenerationMode(Enum):
    PREVIEW = "PREVIEW"
    WRITE = "WRITE"
    STRICT = "STRICT"

    @classmethod
    def from_string(cls, value: str | None) -> AiGenerationMode:
        if value is None or not value.strip():
            return cls.PREVIEW
        normalized = value.strip().upper()
        mapping = {
            "PREVIEW": cls.PREVIEW,
            "WRITE": cls.WRITE,
            "STRICT": cls.STRICT,
        }
        if normalized not in mapping:
            raise ValueError(f"Unsupported mode: {value}. Use preview|write|strict")
        return mapping[normalized]
